rust/agents/scraper/src/date_time.rs

```diff
@@ -0,0 +1,34 @@
+//! A set of tools for working with the sea_orm date/time types.
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use chrono::{Datelike, NaiveDateTime, Timelike};
+use sea_orm::prelude::{TimeDate, TimeDateTime, TimeTime};
+
+pub fn from_date_time_like(dt: &(impl Datelike + Timelike)) -> TimeDateTime {
```

Comment:
It is a pain in the ass to use too. But for what it is worth, I at least see how it came up, because they also have `TimeDate` and `TimeTime` which is starting to show some issues already but it is just a prefix issue.